python/packages/autogen-ext/src/autogen_ext/agents/file_surfer/_file_surfer.py

```diff
@@ -0,0 +1,156 @@
+import json
+import traceback
+from typing import List, Sequence, Tuple
+
+from autogen_agentchat.agents import BaseChatAgent
+from autogen_agentchat.base import Response
+from autogen_agentchat.messages import (
+    ChatMessage,
+    MultiModalMessage,
+    TextMessage,
+)
+from autogen_core.base import CancellationToken
+from autogen_core.components import FunctionCall
+from autogen_core.components.models import (
+    ChatCompletionClient,
+    LLMMessage,
+    SystemMessage,
+    UserMessage,
+)
+
+from ._markdown_file_browser import MarkdownFileBrowser
+
+# from typing_extensions import Annotated
+from ._tool_definitions import (
+    TOOL_FIND_NEXT,
+    TOOL_FIND_ON_PAGE_CTRL_F,
+    TOOL_OPEN_LOCAL_FILE,
+    TOOL_PAGE_DOWN,
+    TOOL_PAGE_UP,
+)
+
+
+class FileSurfer(BaseChatAgent):
+    DEFAULT_DESCRIPTION = "An agent that can handle local files."
```

Comment:
We need some API doc here, including the installation instructions about which extra to use.

Reply:
Yeah, I agree. Still WIP as a I fix remaining issues.

Reply:
Tracking this using #4411